sql/wsrep_binlog.cc

```diff
@@ -22,11 +22,57 @@
 #include "binlog.h"
 #include "log.h"
 #include "log_event.h"  // Log_event_writer
+#include "mutex_lock.h"
 #include "mysql/psi/mysql_file.h"
 #include "service_wsrep.h"
+#include "sql/sql_base.h"  // TEMP_PREFIX
 #include "transaction.h"
 #include "wsrep_applier.h"
-#include "mutex_lock.h"
+
+static bool wsrep_dump_cache_common(
+    IO_CACHE_binlog_cache_storage *cache,
+    std::function<bool(unsigned char *, my_off_t)> dumpFn) {
+  my_off_t const saved_pos(cache->position());
+
+  unsigned char *read_pos = NULL;
+  my_off_t read_len = 0;
+
+  if (cache->begin(&read_pos, &read_len)) {
+    WSREP_ERROR("Failed to initialize io-cache");
+    return true;
+  }
+
+  if (read_len == 0 && cache->next(&read_pos, &read_len)) {
+    WSREP_ERROR("Failed to read from io-cache");
+    return true;
+  }
+
+  bool res = false;
+  while (read_len > 0) {
+    if (dumpFn(read_pos, read_len)) {
+      res = true;
+      break;
+    }
+    cache->next(&read_pos, &read_len);
+  }
+
+  if (cache->truncate(saved_pos)) {
+    WSREP_WARN("Failed to reinitialize io-cache");
+    res = true;
+    ;
```

Comment:
A break or something missing here?

Reply:
No, this is just a not needed semicolon.